updating Windows networking section with new features (#21534)
kubernetes/website: content/en/docs/setup/production-environment/windows/intro-windows-in-kubernetes.md

* Node -> Pod
* Pod -> Node

> ##### IPAM

Reviewer: ```suggestion
##### IP address management (IPAM) {#ipam}
```

The following IPAM options are supported on Windows:

* [Host-local](https://github.com/containernetworking/plugins/tree/master/plugins/ipam/host-local)
* HNS IPAM (Inbox platform IPAM, this is a fallback when no IPAM is set)
* [Azure-vnet-ipam](https://github.com/Azure/azure-container-networking/blob/master/docs/ipam.md) (for azure-cni only)

> ##### Services & Load Balancing
> The following settings applicable to services and load balancing are available on Windows:

Reviewer: nit:
```suggestion
##### Load balancing and Services

On Windows, you can use the following settings to configure Services and load balancing behavior:```

This wording makes it clear that “Service” refers to the Kubernetes resource by that name.

> | Feature | Description | Supported Kubernetes version  | Supported Windows OS build | How to enable |

Author: Added caption

| ------- | ----------- | ----------------------------- | -------------------------- | ------------- |
| Session affinity | Ensures that connections from a particular client are passed to the same Pod each time. | v1.19+ | [Windows Server vNext Insider Preview Build 19551](https://blogs.windows.com/windowsexperience/2020/01/28/announcing-windows-server-vnext-insider-preview-build-19551/) (or higher) | Set `service.spec.sessionAffinity` to "ClientIP" |

> | DSR | Load balancing mode where the IP fixups and the LBNAT occurs at the container vSwitch port directly; service traffic arrives with the source IP set as the originating pod IP. Promises lower latency and scalability. | v1.15+ | Windows Server, version 2004 | Set the following flags in kube-proxy: `feature-gates="WinDSR=true" --enable-dsr=true` |

Reviewer: ```suggestion
| Direct Server Return | Load balancing mode where the IP address fixups and the LBNAT occurs at the container vSwitch port directly; service traffic arrives with the source IP set as the originating pod IP. Promises lower latency and scalability. | v1.15+ | Windows Server, version 2004 | Set the following flags in kube-proxy: `feature-gates="WinDSR=true" --enable-dsr=true` |
```

> | Preserve-DIP | Skips DNAT of service traffic, thereby preserving the virtual IP of the target service in packets reaching the backend Pod. This setting will also ensure that the client IP of incoming packets get preserved. | v1.15+ | Windows Server, version 1903 (or higher) | Set `"preserve-destination": "true"` in service annotations and enable DSR. |

Reviewer: It doesn't seem clear with this row whether the reader needs to reconfigure `kube-proxy` to enable DSR as per the previous row, or enable DSR elsewhere (eg: a Windows setting).

Author: Clarified DSR in kube-proxy is needed. If you have any suggestion on how to convey this more concisely please do share!

| IPv4/IPv6 dual-stack networking | Native IPv4-to-IPv4 in parallel with IPv6-to-IPv6 communications to, from, and within a cluster | v1.19+ | Windows Server vNext Insider Preview Build 19603 (or higher) | See [IPv4/IPv6 dual-stack](#ipv4ipv6-dual-stack) |

#### IPv4/IPv6 dual-stack
IPv4/IPv6 dual-stack networking on Windows can be enabled for `l2bridge` networks using the `IPv6DualStack` feature gate. For more details, see how to [enable IPv4/IPv6 dual stack](../../../../concepts/services-networking/dual-stack#enable-ipv4ipv6-dual-stack).

{{< note >}}
Requires Windows Server vNext Insider Preview Build 19603 (or higher). 

Overlay (VXLAN) networks on Windows do not support dual-stack networking today.

> This feature will become available on Windows Server, version 2004 through a cumulative update tentatively scheduled to be released in 1.19 - 1.20 release timeframe.

Reviewer: Ideally, [avoid making statements about the future](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-statements-about-the-future) in documentation.

It's OK to make those statements in KEPs, blogs, release notes, on Slack etc. In documentation these quickly become stale, and also define a commitment that the Kubernetes project isn't always able to make.

Author: removed this announcement.

* On Linux, you have a DNS suffix list, which is used when trying to resolve PQDNs. On Windows, we only have 1 DNS suffix, which is the DNS suffix associated with that pod's namespace (mydns.svc.cluster.local for example). Windows can resolve FQDNs and services or names resolvable with just that suffix. For example, a pod spawned in the default namespace, will have the DNS suffix **default.svc.cluster.local**. On a Windows pod, you can resolve both **kubernetes.default.svc.cluster.local** and **kubernetes**, but not the in-betweens, like **kubernetes.default** or **kubernetes.default.svc**.
* On Windows, there are multiple DNS resolvers that can be used. As these come with slightly different behaviors, using the `Resolve-DNSName` utility for name query resolutions is recommended.

##### IPv6

> Windows does not support "IPv6-only" networking. However, Windows does support dual-stack IPv4/IPv6 networking for pods and nodes with single-family services. See [IPv4/IPv6 dual-stack networking](#ipv4ipv6-dual-stack) for more details.

Reviewer: _Windows_ can work with single-stack IPv6. Make it clear that this applies to Kubernetes on Windows, rather than the operating system itself.

Author: I reworded this section now.

##### Session affinity

> Setting the maximum session sticky time using `service.spec.sessionAffinityConfig.clientIP.timeoutSeconds` is not supported on Windows as this has not been implemented.

Reviewer: ```suggestion
Setting the maximum session sticky time using `service.spec.sessionAffinityConfig.clientIP.timeoutSeconds` is not implemented for Windows nodes.
```

(what happens if you set it anyway?)
